bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/GenerateAntScriptForMps.kt

```diff
@@ -0,0 +1,141 @@
+            <property name="mps.home" location="build/mps" />
+            <property name="artifacts.mps" location="${"$"}{mps.home}" />
+            <property name="environment" value="env" />
+            <property name="env.JAVA_HOME" value="${"$"}{java.home}/.." />
+            <property name="jdk.home" value="${"$"}{env.JAVA_HOME}" />
+
+            <path id="path.mps.ant.path">
+                ${antLibs.joinToString("\n                    ") {
+                """<pathelement location="${"$"}{artifacts.mps}/$it" />"""
+            }}
+            </path>
+
+            <target name="build" depends="${if (isExport) "export" else "import"}-modules" />
+
+            <target name="clean">
+                <delete dir="${"$"}{build.mps.config.path}" />
+                <delete dir="${"$"}{build.mps.system.path}" />
+            </target>
+
+            <target name="declare-mps-tasks">
+                <taskdef resource="jetbrains/mps/build/ant/antlib.xml" classpathref="path.mps.ant.path" />
+            </target>
+
+            <target name="${if (isExport) "export" else "import"}-modules" depends="clean,declare-mps-tasks">
+                <echo message="Running ${if (isExport) "export" else "import"} of modules" />
+                <runMPS solution="ac6b4971-2a89-49fb-9c30-c2f0e85de741(org.modelix.mps.model.sync.bulk)" startClass="org.modelix.mps.model.sync.bulk.MPSBulkSynchronizer" startMethod="${if (isExport) "export" else "import"}Repository">
+                    <library file="${mpsDependenciesPath.get()}" />
+                    <library file="${repositoryPath.get()}" />
+                    <jvmargs>
+                        <arg value="-Dmodelix.mps.model.sync.bulk.${if (isExport) "output" else "input"}.path=${jsonDirPath.get()}" />
+                        <arg value="-Dmodelix.mps.model.sync.bulk.${if (isExport) "output" else "input"}.modules=${includedModules.get().joinToString(",")}" />
+                        <arg value="-Dmodelix.mps.model.sync.bulk.repo.path=${repositoryPath.get()}" />
+                        <arg value="-Didea.config.path=${"$"}{build.mps.config.path}" />
+                        <arg value="-Didea.system.path=${"$"}{build.mps.system.path}" />
+                        <arg value="-ea" />
+                        <arg value="-Xmx${mpsHeapSize.get()}" />
+                    </jvmargs>
+                </runMPS>
+            </target>
+        </project>
+            """.trimIndent(),
+        )
+    }
+
+    private fun getAntLibs(mpsHome: File): List<String> {
+        val mpsVersion = getMpsVersion(mpsHome)
+        return if (mpsVersion < "2021.2") {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/log4j.jar", "lib/jdom.jar")
+        } else if (mpsVersion < "2021.3") {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/util.jar")
+        } else {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/util.jar", "lib/3rd-party-rt.jar")
+        }
```

Comment:
Can you add some explanation for each of the conditions on why the exact combination was chosen. If there was ever a bug affecting this code, people would easily be lost on why the current solution was used.

Reply:
It's just copy pasted from the api-gen gradle plugin. https://issues.modelix.org/issue/MODELIX-542/Deduplicate-GenerateAntScriptForMps

Reply:
That still makes it hard to understand why different libraries are used. ;-)